scripts/infra/cloud-instance

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+
+INSTRUCT_LAB_CLOUD_CONFIG=${INSTRUCT_LAB_CLOUD_CONFIG:-$HOME/instructlab/cloud-config}
+
+if test -f $INSTRUCT_LAB_CLOUD_CONFIG ; then
+    . $INSTRUCT_LAB_CLOUD_CONFIG
+else
+    echo "Missing required config: $INSTRUCT_LAB_CLOUD_CONFIG"
+    exit 1
+fi
+
+ec2() {
+    local cmdname=$1; shift
+    if type "ec2__$cmdname" >/dev/null 2>&1; then
+        "ec2__$cmdname" "$@"
+    else
+        echo "Invalid command: $cmdname"
+        show_usage
+        exit 1
+    fi
+}
+
+ec2__launch() {
+    while getopts "h?n:k:t:" opt; do
+        case "$opt" in
+            h|\?)
+            show_usage
+            exit 0
+            ;;
+            n)  EC2_INSTANCE_NAME=$OPTARG
+            ;;
+            k)  EC2_KEY_NAME=$OPTARG
+            ;;
+            t)  EC2_INSTANCE_TYPE=$OPTARG
+            ;;
+        esac
+    done
+
+    local instance_id=$(aws ec2 run-instances \
+        --image-id $EC2_AMI_ID \
+        --region $EC2_REGION \
+        --instance-type $EC2_INSTANCE_TYPE \
+        --security-group-ids $EC2_SECURITY_GROUP_ID \
+        --subnet-id $EC2_SUBNET_ID \
+        --key-name $EC2_KEY_NAME \
+        --block-device-mappings '{"DeviceName": "/dev/sda1","Ebs": {"VolumeSize": 300}}' \
+        --associate-public-ip-address \
+        --tag-specifications "ResourceType=instance,Tags=[{Key=Name,Value=$EC2_INSTANCE_NAME}]" \
+        "ResourceType=volume,Tags=[{Key=Name,Value=$EC2_INSTANCE_NAME}]" \
+        --query 'Instances[0].InstanceId' \
+        --output text) &> /dev/null
+
+
+    echo "Waiting for instance to be in the 'running' state..."
+    while state=$(aws ec2 describe-instances \
+                --instance-ids $instance_id \
+                --region $EC2_REGION \
+                --query 'Reservations[0].Instances[0].State.Name' \
+                --output text) &> /dev/null; [ "$state" != "running" ]; do
+        sleep 5
+    done
+
+    echo "Instance is now running."
```

Comment:
Print instance ID?

Reply:
Part of the intention of this utility is to not have to deal with the instance id.  Printing it out might make the user think it was needed for something.

Reply:
Changed it to print the name and id next to it in ()